rest/server_context.go

```diff
@@ -509,6 +511,17 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 	if dbName == "" {
 		dbName = spec.BucketName
 	}
+	defer func() {
+		if returnedError != nil {
+			_, dbRegistered := sc.databases_[dbName]
+			if !dbRegistered && dbcontext != nil {
+				dbcontext.Close(ctx) // will close bucket
+			} else if bucket != nil {
+				bucket.Close(ctx)
```

Comment:
In the case where dbRegistered==true, I don't think it's safe to close the bucket either, is it?